src/server/game/Spells/Spell.cpp

```diff
@@ -5831,46 +5831,70 @@ SpellCastResult Spell::CheckRange(bool strict)
     if (!strict && m_casttime == 0)
         return SPELL_CAST_OK;
 
+    float minRange, maxRange;
+    std::tie(minRange, maxRange) = GetMinMaxRange(strict);
+
+    // get square values for sqr distance checks
+    minRange *= minRange;
+    maxRange *= maxRange;
+
     Unit* target = m_targets.GetUnitTarget();
+    if (target && target != m_caster)
+    {
+        if (m_caster->GetExactDistSq(target) > maxRange)
+            return SPELL_FAILED_OUT_OF_RANGE;
+
+        if (minRange > 0.0f && m_caster->GetExactDistSq(target) < minRange)
+            return SPELL_FAILED_OUT_OF_RANGE;
+
+        if (m_caster->GetTypeId() == TYPEID_PLAYER &&
+            (m_spellInfo->FacingCasterFlags & SPELL_FACING_FLAG_INFRONT) && !m_caster->HasInArc(static_cast<float>(M_PI), target))
+            return SPELL_FAILED_UNIT_NOT_INFRONT;
+    }
+
+    if (m_targets.HasDst() && !m_targets.HasTraj())
+    {
+        if (m_caster->GetExactDistSq(m_targets.GetDstPos()) > maxRange)
+            return !(_triggeredCastFlags & TRIGGERED_DONT_REPORT_CAST_ERROR) ? SPELL_FAILED_OUT_OF_RANGE : SPELL_FAILED_DONT_REPORT;
+        if (minRange > 0.0f && m_caster->GetExactDistSq(m_targets.GetDstPos()) < minRange)
+            return !(_triggeredCastFlags & TRIGGERED_DONT_REPORT_CAST_ERROR) ? SPELL_FAILED_OUT_OF_RANGE : SPELL_FAILED_DONT_REPORT;
+    }
+
+    return SPELL_CAST_OK;
+}
+
+std::pair<float, float> Spell::GetMinMaxRange(bool strict)
+{
+    float rangeMod = 0.0f;
     float minRange = 0.0f;
     float maxRange = 0.0f;
-    float rangeMod = 0.0f;
-    if (strict && IsNextMeleeSwingSpell())
+
+    if (strict && IsNextMeleeSwingSpell()) 
```

Comment:
I think it improves readability, but it would be nice to have consistency in the spacing.
I would remove the blank line before `return SPELL_CAST_OK;` instead (Line 5862).
Maybe make it a rule to have a blank line before each `if ()` check or no extra LF at all?